dist/extension.js

```diff
@@ -0,0 +1,47 @@
+const vscode = require('vscode');
+
+const THEME_CONFIGURATION_ID = 'markdown-preview-github-styles';
+const THEME_CONFIGURATION_KEY = 'colorMode';
+
+const THEME_CONFIGURATION_VALUES = {
+    'auto': true,
+    'system': true,
+    'light': true,
+    'dark': true,
+    'high-contrast': true
+}
+const THEME_CONFIGURATION_AUTO = 'auto';
+
+let settings = vscode.workspace.getConfiguration(THEME_CONFIGURATION_ID, null);
+
+function getColorMode() {
+    return validThemeConfigurationValue(settings.get(THEME_CONFIGURATION_KEY))
+}
+
+function validThemeConfigurationValue(theme) {
+    return !THEME_CONFIGURATION_VALUES[theme]
+        ? THEME_CONFIGURATION_AUTO
+        : theme;
+}
+
+exports.activate = function() {
+    vscode.workspace.onDidChangeConfiguration(function(e) {
+        if (e.affectsConfiguration(`${THEME_CONFIGURATION_ID}.${THEME_CONFIGURATION_KEY}`)) {
+            settings = vscode.workspace.getConfiguration(THEME_CONFIGURATION_ID, null);
+        }
+    });
+
+    return {
+        extendMarkdownIt(md) {
+            return md.use(exports.plugin);
+        }
+    };
+}
+
+exports.plugin = function(md) {
```

Comment:
[nit] Does this need to be exported?

Reply:
Was thinking of test cases but privatised as of b2c8012